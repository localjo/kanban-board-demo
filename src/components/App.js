import React from 'react';
import Header from './layout/Header';
import Board from './screens/Board';

function App() {
  return (
    <>
      <Header />
      <Board />
    </>
  );
}

export default App;
