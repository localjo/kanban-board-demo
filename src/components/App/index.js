import React from 'react';

import useLocalStorage from '../../utils/useLocalStorage';
import Header from '../Header';
import Board from '../Board';
import Card from '../Card';
import Column from '../Column';

function App() {
  const [cards, setCards] = useLocalStorage({
    'To do': [
      { title: 'Three', description: 'Test card 3', email: 'me@example.com' },
    ],
    'In progress': [
      { title: 'Two', description: 'Test card 2', email: 'me@example.com' },
    ],
    Done: [
      { title: 'One', description: 'Test card 1', email: 'me@example.com' },
    ],
  });
  const columns = Object.keys(cards);
  return (
    <>
      <Header />
      <Board>
        {columns.map(column => {
          return (
            <Column key={column} title={column}>
              {cards[column].map(card => (
                <Card {...card} key={card.title} />
              ))}
            </Column>
          );
        })}
      </Board>
    </>
  );
}

export default App;
