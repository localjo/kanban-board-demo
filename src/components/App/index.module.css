.appContainer {
  display: flex;
  flex-flow: column;
  align-items: stretch;
  height: 100vh;
}
