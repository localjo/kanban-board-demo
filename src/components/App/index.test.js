import React from 'react';
import ReactDOM from 'react-dom';
import App, { addCard } from '.';

const card1 = {
  title: 'Card One',
  user: 'example@example.com',
  description: 'Test card',
  column: 'To do',
};

describe('App Wrapper', function() {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('correctly adds a card to the state', function() {
    const state = {
      'To do': [],
      'In progress': [],
      Done: [],
    };
    expect(addCard(state, card1)).toMatchSnapshot();
  });
});
