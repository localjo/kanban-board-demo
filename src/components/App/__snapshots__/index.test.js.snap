// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Wrapper correctly adds a card to the state 1`] = `
Object {
  "Done": Array [],
  "In progress": Array [],
  "To do": Array [
    Object {
      "column": "To do",
      "description": "Test card",
      "title": "Card One",
      "user": "example@example.com",
    },
  ],
}
`;

exports[`App Wrapper correctly sorts cards in the state 1`] = `
Object {
  "Done": Array [],
  "In progress": Array [],
  "To do": Array [
    Object {
      "column": "To do",
      "description": "Test card",
      "title": "Card Two",
      "user": "example@example.com",
    },
    Object {
      "column": "To do",
      "description": "Test card",
      "title": "Card Three",
      "user": "example@example.com",
    },
    Object {
      "column": "To do",
      "description": "Test card",
      "title": "Card One",
      "user": "example@example.com",
    },
  ],
}
`;
